### ISSUE TITLE
Doesn't compile on MacOS:  wrong number of arguments for 'LibC::PthreadT#to_s'

### ISSUE THREAD

**Issue by spuun** [ysbaddaden/execution_context#12]
I'm trying to compile on MacOS but i'm getting this error:

```
In lib/execution_context/src/isolated.cr:150:29

 150 | thread.@system_handle.to_s(io, 16)
                             ^---
Error: wrong number of arguments for 'LibC::PthreadT#to_s' (given 2, expected 0..1)

Overloads are:
 - Pointer(T)#to_s(io : IO)
 - Object#to_s()
```

IIUC `PthreadT` is `Void*` (type, not alias) while it's `ULong` on some other OS:es (e.g. linux), and it may be the culprit?

**Comment by ysbaddaden** [comment]
Oh right, it's only an unsigned long on linux (btw: the musl-libc bindings wrongly define it as a void pointer). I should add a BSD VM on my computer.

**Comment by ysbaddaden** [comment]
Should be fixed in 67ac1d0e61ec3baeec68942805396d5b1551a776 where I removed the thread handle from `#inspect` and take care of the linux case in `Crystal.trace` (for `-Dsched_tracing` flag to trace what's happening on STDERR).
